dttools/src/rmsummary.c

```diff
@@ -699,18 +699,29 @@ void rmsummary_bin_op(struct rmsummary *dest, const struct rmsummary *src, rm_bi
 }
 
 /* Copy the value for all the fields in src > -1 to dest */
-static double override_field(double d, double s)
+static inline double override_field(double d, double s)
 {
 	return (s > -1) ? s : d;
 }
 
+/* This function may be called many times, therefore it is worthwhile to omit/inline
+ * function calls and remove indirect jumps to rmsummary_bin_op */
 void rmsummary_merge_override(struct rmsummary *dest, const struct rmsummary *src)
 {
 	if(!src) {
 		return;
 	}
 
-	rmsummary_bin_op(dest, src, override_field);
```

Comment:
Instead of this, what about declaring `rmsummary_bin_op` as static inline?

Reply:
Let's have a look at the performance effect before reaching a conclusion..

Reply:
If the call to `rmsummary_bin_op` is really a problem, I would first convert it to a macro rather than copy its contents where it is needed. (I.e., copy the contents with a macro instead.)

Also, the `rmsummary_bin_op` compares more than just cores, memory, disk and gpus.  You may want to add a version that just compares the resources needed in taskvine, as I think we do in category.h.